### DISCUSSION TITLE
Is it possible to combine the "set" and hideKBM attributes?

### DISCUSSION THREAD

**Discussion by TheEmbracedOne** [Orange-Panda/Rewired-Glyphs#14] [Q&A] [answered]
What I'm trying to achieve here is probably possible, but I'm running into an issue. I'll describe my use case:

- I have two glyph collections. One, called "nobg", is without backgrounds, and another set called "bg" has backgrounds baked into glyphs.
- By default, I'd like to display my glyphs without a background, so I would load the nobg collection as the default glyph collection, and put the "bg" one as an additional collection.
- In almost all scenarios where I display glyphs in my game, I only want to display them on gamepad only (this is because you can navigate and do everything on a mouse only and by default there are no keybindings for those actions anyway), for example here on the top navigation bar of the settings screen (it's a bit hard to see but the glyphs here have a tilted rectangle background):
![image](https://github.com/Orange-Panda/Rewired-Glyphs/assets/16197906/a0ce7aa5-1773-491a-a0b7-ff5d623914d8)

The problem I am running into is that I'd like to specify the few places where I want to show glyphs with backgrounds, but I'd also like the option to additionally specify where I want glyphs to only display on gamepad. In short, I have use cases where I would like to use combinations like
- `<Glyph Cancel set=bg hideKBM>` (to display the cancel glyph with a background, gamepad only)
- `<Glyph Cancel hideKBM>` (to display the cancel glyph without a background, gamepad only)

and this is my issue:
![image](https://github.com/Orange-Panda/Rewired-Glyphs/assets/16197906/2aa8696b-2aa0-48ca-8bb6-939c150d79a6)

Additionally, I tried to put no default collection or additional collections at all on the Glyph Manager and just load them via a script like so:
```cs
InputGlyphs.LoadGlyphCollection(noBgCollection, true);
InputGlyphs.LoadGlyphCollection(bgCollection);
```

I also tried making a third glyph collection to load as default, and then load the nobg collection on Start via a script so it does not act as a default collection:

```cs
InputGlyphs.LoadGlyphCollection(noBgCollection);
InputGlyphs.LoadGlyphCollection(bgCollection);
```

but the results were the same unfortunately, with "[Right Mouse Button]" displaying. How could I avoid this please? Thank you

**Comment by Orange-Panda** [comment]
Your glyph tag looks correct so this might be an internal bug. I'll take a look and get back to you with my findings

**Reply by Orange-Panda** [reply, marked as answer]
This was indeed an internal bug. When a glyph map didn't have a glyph it was creating a new glyph as a fallback but that fallback glyph was not considered a `Keyboard` glyph. This has been fixed and I have published a new release `2.1.0` with the fix included.

It should work out of the box without any changes on your end

**Reply by TheEmbracedOne** [reply]
Wonderful, thank you so much!

**Reply by TheEmbracedOne** [reply]
Just wanted to add, now it works perfectly!
